front/components/agent_builder/capabilities/mcp/utils/formStateUtils.ts

```diff
@@ -0,0 +1,61 @@
+import type { UseFormReturn } from "react-hook-form";
+
+import type { MCPFormData } from "@app/components/agent_builder/AgentBuilderFormContext";
+import type { AgentBuilderAction } from "@app/components/agent_builder/types";
+import logger from "@app/logger/logger";
+import type { MCPServerViewType } from "@app/lib/api/mcp";
+
+import { getDefaultFormValues } from "./formDefaults";
+
+/**
+ * Creates form reset function for different dialog modes
+ * @param configurationTool - Current configuration tool (if any)
+ * @param mcpServerView - Current MCP server view (if any)
+ * @param isOpen - Whether the dialog is open
+ * @returns Reset function that accepts form instance when called
+ */
+export function createFormResetHandler(
+  configurationTool: AgentBuilderAction | null,
+  mcpServerView: MCPServerViewType | null,
+  isOpen: boolean
+) {
+  return (form: UseFormReturn<MCPFormData>) => {
+    try {
+      if (!isOpen) {
+        // Dialog closed: reset to empty state and clear errors
+        form.reset(getDefaultFormValues(null));
+        form.clearErrors();
```

Comment:
This would be redundant as reset also clear the error https://react-hook-form.com/faqs#:~:text=React%20Hook%20Form%27s%20reset%20method%20will%20reset%20all%20field%20values%2C%20and%20will%20also%20clear%20all%20errors%20within%20the%20form